.github/workflows/release.yaml

```diff
@@ -1,64 +1,80 @@
+        uses: radcortez/project-metadata-action@master
+        id: metadata
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          metadata-file-path: '.github/project.yaml'
+      - name: Set up Node.js v12
         uses: actions/setup-node@v1
         with:
           node-version: 12
       - name: Set up JDK 1.8
         uses: actions/setup-java@v1
         with:
           java-version: 1.8
+      - name: Set up settings.xml
+        run: |
+          echo "<settings><servers><server><id>${{ secrets.OSSRH_ID }}</id><username>${{ secrets.OSSRH_USERNAME }}</username><password>${{ secrets.OSSRH_TOKEN }}</password></server></servers><profiles><profile><id>${{ secrets.OSSRH_ID }}</id><activation><activeByDefault>true</activeByDefault></activation><properties><gpg.executable>gpg</gpg.executable><gpg.passphrase>${{ secrets.GPG_PASSPHRASE}}</gpg.passphrase></properties></profile></profiles></settings>" > /home/runner/.m2/settings.xml
       - name: Set up Gren
-        if: github.event_name == 'repository_dispatch'
         run: npm install github-release-notes -g
       - name: Cache Dependencies
-        if: github.event_name != 'repository_dispatch'
         uses: actions/cache@v1
         with:
           path: ~/.m2/repository
           key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: |
             ${{ runner.os }}-maven-
-      - name: Build Project
-        if: github.event_name != 'repository_dispatch'
-        run: mvn clean install
-      - name: Setup settings.xml
-        if: github.event_name == 'repository_dispatch'
+      - name: Code Checkout
         run: |
-          echo "<settings><servers><server><id>${{ secrets.OSSRH_ID }}</id><username>${{ secrets.OSSRH_USERNAME }}</username><password>${{ secrets.OSSRH_TOKEN }}</password></server></servers><profiles><profile><id>${{ secrets.OSSRH_ID }}</id><activation><activeByDefault>true</activeByDefault></activation><properties><gpg.executable>gpg</gpg.executable><gpg.passphrase>${{ secrets.GPG_PASSPHRASE}}</gpg.passphrase></properties></profile></profiles></settings>" > /home/runner/.m2/settings.xml
+          git init
+          git config --global user.name "github-actions"
+          git config --global user.email "github-actions@github.com"
+          git remote add origin "https://$GITHUB_ACTOR:${{ secrets.ACCESS_TOKEN }}@github.com/$GITHUB_REPOSITORY.git"
+          git fetch
+          git checkout master
+          git branch --set-upstream-to=origin/master
+          git pull
+          echo "#### Listing files after clone ####"
+          ls -a
+      - name: Update Release Version ${{steps.metadata.outputs.release-version}}
+        run: |
+          mvn versions:set -DnewVersion=${{steps.metadata.outputs.release-version}} -DgenerateBackupPoms=false -DprocessAllModules=true
+          git add -A
+          git commit -m"Automated version update: ${{steps.metadata.outputs.release-version}}"
+          git push
```

Comment:
We need to do a full maven build/verify after we've updated the version # but before we commit that change back.

Reply:
Sure, will do that. Any specific reason we wanna do a build here ?

We anyways build our master branch on every push, so i thought a build not required here. is there a chance of build failure on just version update ? Also, 'mvn clean deploy' will anyways do a build before it publishes to the central repository. If the build fails, there will not be any release either to maven or github. Hence, the reason why I perform a github release after we successfully publish our artifacts to the central repository. Let me know if I am missing a point here.

Reply:
I want to make sure we do a full build/test before the workflow **commits** anything back to the repo.  I don't expect a version change to cause a build failure, but it's better to be sure. :)

Reply:
addressed